src/hooks/useFirestoreCollectionWithAtom.ts

```diff
@@ -190,6 +195,10 @@ export function useFirestoreCollectionWithAtom<T = TableRow>(
               available: docs.length >= memoizedQuery.limit,
             }));
           }
+          // on each new snapshot, use the query to get and set the document count from the server
+          getCountFromServer(memoizedQuery.unlimitedQuery).then((value) => {
+            setServerDocCountAtom(value.data().count)
+          })
```

Comment:
```suggestion
          // on each new snapshot, use the query to get and set the document count from the server
          if (serverDocCountAtom) {
            getCountFromServer(memoizedQuery.unlimitedQuery).then((value) => {
              setServerDocCountAtom(value.data().count)
            })
          }
```
⚠️ `serverDocCountAtom` is an optional prop and we don’t query the total number of docs by default. As said in the comment on lines 107–108, you need to make sure that that prop was passed in the first place.

Currently, if the component calling this hook does not pass that atom, it will overwrite the `dataAtom` with the number of docs in the query, breaking the hook entirely.

We need to pass any atom to `useSetAtom`, otherwise Jotai will throw an error. So that’s why we need to be very careful when making a modification to a hook in this way.